godel_surface_detection/src/nodes/robot_auto_blending.cpp

```diff
@@ -13,87 +11,112 @@
-      ros::ServiceClient surface_client_ = nh.serviceClient<godel_msgs::SurfaceDetection>(SURFACE_DETECTION_SERVICE);
-
-      if (!surface_client_.call(surfacedetection_srv))
-      {
-        ROS_WARN_STREAM("Unable to call surface detection service");
-      }
+      ROS_WARN_STREAM("Unable to call surface_blending_parameters service");
+      serviceCall_failure++;
+      continue;
+    }
 
-      //Select all surface
-      godel_msgs::SelectSurface::Request req;
-      godel_msgs::SelectSurface::Response res;
-      req.action = req.SELECT_ALL;
-      ros::ServiceClient select_surface_client_ = nh.serviceClient<godel_msgs::SelectSurface>(SELECT_SURFACE_SERVICE);
-      bool succeeded = select_surface_client_.call(req, res);
+    //Scan and detect surface
+    surfacedetection_srv.request.action = 3;
+    surfacedetection_srv.request.use_default_parameters = false;
+    surfacedetection_srv.request.robot_scan = param_srv.response.robot_scan;
+    surfacedetection_srv.request.surface_detection = param_srv.response.surface_detection;
+    if (!surface_client_.call(surfacedetection_srv))
+    {
+      ROS_WARN_STREAM("Unable to call surface detection service");
+      serviceCall_failure++;
+      continue;
+    }
 
-      //Generate path
-      godel_msgs::ProcessPlanning process_plan;
-      process_plan.request.use_default_parameters = false;
-      process_plan.request.params = param_srv.response.blending_plan;
-      process_plan.request.scan_params =param_srv.response.scan_plan;
-      process_plan.request.action = 2;
-      ros::ServiceClient process_plan_client_ = nh.serviceClient<godel_msgs::ProcessPlanning>(PROCESS_PATH_SERVICE);
+    //Select all surface
+    req.action = req.SELECT_ALL;
+    if (!select_surface_client_.call(req, res))
+    {
+      ROS_WARN_STREAM("Unable to call select_surface service");
+      serviceCall_failure++;
+      continue;
+    }
 
-      if (process_plan_client_.call(process_plan))
-      {
-        godel_msgs::GetAvailableMotionPlans motionQuery_srv;
-        ros::ServiceClient get_motion_plans_client_ = nh.serviceClient<godel_msgs::GetAvailableMotionPlans>(GET_AVAILABLE_MOTION_PLANS_SERVICE);
-        if (get_motion_plans_client_.call(motionQuery_srv))
-        {
-          plan_names = motionQuery_srv.response.names;
-        }
-        else
-		{
-          ROS_ERROR_STREAM("Could not get names from 'available motions server'");
-        }
-      }
+    //Generate path
+    process_plan.request.use_default_parameters = false;
+    process_plan.request.params = param_srv.response.blending_plan;
+    process_plan.request.scan_params = param_srv.response.scan_plan;
+    process_plan.request.action = 2;
+    if (!process_plan_client_.call(process_plan))
+    {
+      ROS_WARN_STREAM("Unable to call process_path service");
+      serviceCall_failure++;
+      continue;
+    }
+    if (!get_motion_plans_client_.call(motionQuery_srv))
```

Comment:
A blank line separating these two if statements wouldn't hurt, although that is just a personal style preference